cubeviz/controls/units_ui.py

```diff
@@ -0,0 +1,202 @@
+from __future__ import absolute_import, division, print_function
+
+from qtpy.QtCore import Qt
+from qtpy import QtGui
+from qtpy.QtWidgets import (
+    QDialog, QApplication, QPushButton,
+    QLabel, QWidget, QHBoxLayout, QVBoxLayout, QLineEdit, QComboBox
+)
+
+from astropy.stats import sigma_clip
+import numpy as np
+import re
+
+class WavelengthUI(QDialog):
+    def __init__(self, wavelength_controller, parent=None):
+        super(WavelengthUI,self).__init__(parent)
+
+        self.setWindowTitle("Wavelengths")
+
+        self.setWindowFlags(self.windowFlags() | Qt.Tool)
+        self.title = "Wavelengths"
+
+        self._general_description = "From this dialog one may choose the wavelength units and/or display the Rest Wavelength units based on a red-shift Z value."
+
+        self.wavelength_controller = wavelength_controller
+        self.currentAxes = None
+        self.currentKernel = None
+
+        self.createUI()
+
+    def createUI(self):
+        """
+        Create the popup box with the calculation input area and buttons.
+
+        :return:
+        """
```

Comment:
I've mentioned this before but I think it would be a good idea in the long term if we redefined all of these UI layouts in `.ui` files using Qt Designer.

Reply:
agreed. we can do that part of a cleanup.